integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+import { getTestName } from "../names";
+import {
+  v1ApiPath,
+  saveCreds,
+  getCreds,
+  uploadFileRequest,
+  checkRequestFails,
+  makeAuthorizedRequest,
+  makeAuthorizedRequestWithStatus,
+  sortArrayOn,
+  sortArrayOnTwoKeys
+} from "../server";
+import { logTestDescription } from "../descriptions";
+const url = require("url");
+
+Cypress.Commands.add(
+  "recordEvent",
+  (
+    camera: string,
+    type: string,
+    details = {},
+    date = new Date(),
+    log = true
+  ) => {
+    const data = {
+      dateTimes: [date.toISOString()],
+      description: { type: type, details: details }
+    };
+    logTestDescription(
+      `Create ${type} event for ${camera} at ${date}`,
+      { data: data },
+      log
+    );
+    makeAuthorizedRequest(
+      {
+        method: "POST",
+        url: v1ApiPath("events"),
+        body: data
+      },
+      camera
+    );
+  }
+);
+
+Cypress.Commands.add(
+  "apiCreateCamera",
+  (cameraName: string, group: string, saltId: number = null, log = true, statusCode: number = 200) => {
+    logTestDescription(
+      `Create camera '${cameraName}' in group '${group}' with saltId '${saltId}'`,
+      {
+        camera: cameraName,
+        group: group,
+	saltId: saltId
+      },
+      log
+    );
+
+    const request = createCameraDetails(cameraName, group, null, saltId);
+    if(statusCode==200) {
+      cy.request(request).then((response) => {
+        const id = response.body.id;
+        saveCreds(response, cameraName, id);
+      });
+    } else {
+      checkRequestFails(request);
+    };
+  }
+);
+
+Cypress.Commands.add(
+  "apiDeviceReregister",
+  (oldName: string, newName: string, newGroup: string, password: string = null, keepName:boolean = false, statusCode: number = 200) => {
```

Comment:
No. keepName is used to disable generating a unique name by appending a prefix (so we can test invalid name strings).    Have renamed to doNotMakeNameUnique for clarity

Reply:
No - nt what keepName was used for. keepName was used by clare to disable unique name generation (so we could test specific strings in the name field).  I have renamed the field to generateUniqueName (and reversed the logic) to avoid future confusion.